### ISSUE TITLE
[axe-core warning] - Ensure List has an accessible name

### ISSUE THREAD

**Issue by pranav300** [cerner/terra-core#2623]
# Bug Report

## Description
Having axe-core violations for listItem in terra-list.

## Steps to Reproduce

1. Go to https://engineering.cerner.com/terra-core/tests/terra-list/list/list-item
2. run axe-core on the page.

## Additional Context / Screenshots

![Screenshot 2019-09-04 at 3 23 24 PM](https://user-images.githubusercontent.com/29273628/64245276-3d900180-cf28-11e9-871e-f5e96dc4eafa.png)

## Expected Behavior

There should be axe violations.

## Possible Solution

Add prop ariaLabel to the ListItem

## @ Mentions

@ryanthemanuel @bjankord

**Comment by pranav300** [comment]
**Investigation:**

Whenever a `List` contains a `ListItem` that is selectable, I notice that the corresponding `<ul>` element is an ARIA input field which in-turn means that the `<ul>` element might need to have `aria-label` according to the [documentation](https://dequeuniversity.com/rules/axe/3.3/aria-input-field-name?application=AxeChrome).

So I tested by adding `ariaLabel` to our `List` element and it I found no accessibility violation.
@bjankord What are your thoughts on it?

**Comment by venkateshan6543** [comment]
We are facing the same issue in our Provider-relationship-js project and adding aria-label fixed the accessibility violation.

**Comment by bjankord** [comment]
We can get this updated in our test example. For teams seeing this issue, the recommended fix is to add an aria-label to your list that describes the list.

**Comment by pranav300** [comment]
Had an offline conversation with @dkasper-was-taken to update code example and guides for `SingleSelectList` and `MultiSelectList`.
